pandas/core/generic.py

```diff
@@ -3635,9 +3635,17 @@ def to_csv(
             sequence should be given if the object uses MultiIndex. If
             False do not print fields for index names. Use index_label=False
             for easier importing in R.
-        mode : str, default 'w'
-            Python write mode. The available write modes are the same as
-            :py:func:`open`.
+        mode : {{'w', 'x', 'a'}}, default 'w'
+            Forwarded to either `open(mode=)` or `fsspec.open(mode=)` to control
+            the file opening. Typical values include:
+
+            - 'w', open for writing, truncating the file first.
+            - 'x', open for exclusive creation, failing if the file already exists.
+            - 'a', open for writing, appending to the end of file if it exists.
+
+            Including 'b' or 't' in the mode parameter will inform Pandas whether
+            `path_or_buf` requires string or binary data. However, in most cases,
+            this should not be necessary.
```

Comment:
This is a good point, but I don't think it makes sense to have binary csv files. I think this was a comment in the issue, about using `b`, which in general makes sense, but I don't see how for `to_csv` it could make sense to save data in a binary file. Unless I'm missing something (correct me if I'm wong), better to simply delete this.

Reply:
Thank you for your input.

I also think the same, but we had a discussion with @twoertwein earlier here https://github.com/pandas-dev/pandas/pull/51881#discussion_r1132568198.

Could you please share your thoughts on this?

Reply:
If that part is more confusing than helpful, feel free to remove it

Reply:
Thanks for clarifying. I agree with these comments in general, bit I don't think in this particular case it's possible to binary data. Csv relies on the delimiter, the quote character and the line break to understand the structure. Binary data could have any of those and break a csv. Unless I'm missing something, it's not that in most cases "b" shouldn't be used, it can never be used. To me it makes more sense to remove that comment, since I think it's giving the impression that users can actually use binary csv's in some cases, and some may start researching about it. So, probably a bit confusing and distracting.

Feel free to disagree, but that's how I feel about that comment.